modules/storage-expanding-flexvolume.adoc

```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies
+//
+// * storage/expanding-persistent-volumes.adoc
+
+[id="expanding-flexvolume_{context}"]
+= Expanding FlexVolume with a supported driver
+
+When using FlexVolume to connect to your backend storage system, you can expand persistent storage volumes after they have already been created. This is done by manually updating the persistent volume claim (PVC) in {product-title}.
+
+FlexVolume allows expansion if the driver is set with `RequiresFSResize` to `true`. The FlexVolume can be expanded on Pod restart.
+
+Similar to other volume types, FlexVolume volumes can also be expanded when in use by a Pod.
+
+.Prerequisites
+
+* The underlying volume driver supports resize.
+* The driver is set with the `RequiresFSResize` capability to `true`.
+* Dynamic provisioning is used.
+* The controlling StorageClass has `allowVolumeExpansion` set to `true`.
+* The `PersistentVolumeClaimResize` admission plugin is enabled (optional).
+
+.Procedure
+
+* To use resizing in the FlexVolume plugin, you must implement the `ExpandableVolumePlugin` interface using these methods:
+
+`ExpandVolumeDevice`::
```

Comment:
Since Openshift does not support installation of flexvolume plugins on master nodes - this documentation is not accurate.

```
To use resizing in the FlexVolume plugin, you must implement the `NodeExpandableVolumePlugin` interface using these methods:

RequiresFSResize() bool::
Which indicates that the volume supports expansion on the node.

`NodeExpand`:
This function will internally call `expandfs` on flex volume
```

Since Openshift doed not support installation of flexvolume plugins on master nodes, Openshift does not support control-plane expansion of flexvolume.

Reply:
@gnufied If I understand correctly, when you say "OpenShift does not support control-plane expansion of flexvolume" means it does not support the text only in the first method (`ExpandVolumeDevice`). Or do you mean the other two methods (`RequiresFSResize` and `ExpandFS`) are also not supported?